Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -573,8 +573,12 @@ void RenderObject::clearNeedsLayout(HadSkippedLayout hadSkippedLayout)
     setEverHadLayout();
     setHadSkippedLayout(hadSkippedLayout == HadSkippedLayout::Yes);
 
-    if (auto* renderElement = dynamicDowncast<RenderElement>(*this))
+    if (auto* renderElement = dynamicDowncast<RenderElement>(*this)) {
         renderElement->setLayoutIdentifier(renderElement->view().frameView().layoutContext().layoutIdentifier());
+
+        if (hasLayer())
+            downcast<RenderLayerModelObject>(*this).layer()->setSelfAndChildrenNeedPositionUpdate();
```

Comment:
It's a bit odd that here at _clear_ we unconditionally mark the layer with selfAndChildrenNeedsPositionUpdate regardless of what actually happened.

Reply:
> I mostly modelled it off the setLayoutIdentifier call above, which is a pseudo-dirty-bit for layer tree updates.

Not sure how much that code changed, butI initially the simplified layout dirty bit drove the update flow and the layout id was only used to identify which renderer participated in the last layout.

styleWill/DidChange may be a better candidate though I am not sure if certain tree mutation should also trigger setSelfAndChildrenNeedPositionUpdate.